API-GUIDELINES.md

```diff
@@ -0,0 +1,54 @@
+# esp-rs API Guidelines
+
+## About
+
+This is a living document - make sure to check the latest version of this document.
+
+> [!NOTE]
+> Not all of the currently existing code follows this guideline, yet.
+
+In general, the [Rust API Guidelines](https://rust-lang.github.io/api-guidelines) apply to all projects in the ESP-RS GitHub organization where possible. (`C-RW-VALUE` and `C-SERDE` do not apply) 
+
+Especially for public API but if possible also for internal APIs.
+
+The following paragraphs contain additional recommendations.
+
+## Construction and Destruction of Drivers
+
+- Drivers take peripherals and pins via the `PeripheralRef` pattern - they don't consume peripherals/pins
```

Comment:
This is a little confusing, and I'm not entirely sure what it's trying to convey to users, I don't think it's fair to assume they're familiar with this pattern

Reply:
mhhh would a link to https://docs.esp-rs.org/esp-hal/esp-hal/0.17.0/esp32c6/esp_hal/peripheral/struct.PeripheralRef.html help? (But the problem is we currently need to update the link with every release 🤔  )

Not sure how to better explain the pattern ... but contributors should definitely make themselves familiar with it